### ISSUE TITLE
Endpoint::connect_lazy should not return a Result<>

### ISSUE THREAD

**Issue by nbaztec** [hyperium/tonic#708]
### Crates

`transport`

### Motivation

Most of the object creation and error validation is already done in `Endpoint::new` and as such `Endpoint::connect_lazy` should not be returning any errors since the connection is evaluated on-demand and cannot fail. Meanwhile the API gives the wrong impression that the call may fail (does it ping the service beforehand? who knows)

### Proposal

Change the return signature for `connect_lazy` to return the `Channel` to better indicate via API that this call can not fail due to connection errors.

### Alternatives

Add a doc comment specifying that this function never fails when invoked.

**Comment by davidpdrsn** [comment]
I have had the same thought. @LucioFranco do you know if there is a reason it currently returns a result? Future proofing perhaps? 🤷

**Comment by LucioFranco** [comment]
Yeah, this should have been fixed in 0.5. For now the API is unfalliable and unwraping there is safe from a panic. We should remove it for the next iteration.
